tiledb/sm/c_api/tiledb.h

```diff
@@ -934,10 +934,11 @@ TILEDB_EXPORT void tiledb_config_free(tiledb_config_t** config);
  *    parallel.<br>
  *    **Default**: 1
  * - `sm.num_tbb_threads` <br>
```

Comment:
Do we need another config name, or just add a new config to avoid confusion for those not using TBB? That's a long-term issue that we will have to eventually address in the docs as well.

Reply:
This is a good question so let's discuss this before I go back to your other comments:
My intention was to let this thread pool be relatively hidden/undocumented until we decide on the scope of the thread pool. This is why I'm reusing the config and using it within the `tbb_state.cc`. For instance, we no longer have the limitation that the TBB scheduler is used for the entire process. They could be per-context. Additionally, we may want to refer to as something else: maybe "cpu threadpool" and introduce another "io threadpool".

I think the next steps should be:
1. Benchmark/optimize until we're confident it is consistently better than TBB.
2. Determine the scope of this, and if we should have an associated IO threadpool.
3. Default-off TBB.
4. Remove TBB.

Reply:
Agreed, good plan with one addition:

5. Clean up code comments/functions and document everything! :)

Reply:
I like the idea of having two "high level" thread options, one for IO and one for this threadpool. I can see the use case in having the lower level configs for maximum configurability but in my opinion they should default to being unused and the high level "simple" config parameters are what most people can use to tweak things.

Reply:
I totally agree - we should expose a "CPU" (or any other name here) thread pool (for compression, sorting, result shuffling, etc) and an "IO" pool used exclusively for IO. That would make it easier for the user to reason about parallelism and of course it'll be easier to document.

Reply:
Sounds good to me, we'll go that route. I'll split up these 5 work items into clubhouse tickets. In the mean time, we'll leave references to tbb/global-tp as-is.